rustls/src/client/client_conn.rs

```diff
@@ -86,20 +86,28 @@ pub trait ClientSessionStore: Send + Sync {
 /// A trait for the ability to choose a certificate chain and
 /// private key for the purposes of client authentication.
 pub trait ResolvesClientCert: Send + Sync {
-    /// With the server-supplied acceptable issuers in `acceptable_issuers`,
-    /// the server's supported signature schemes in `sigschemes`,
-    /// return a certificate chain and signing key to authenticate.
+    /// Optionally resolve a client certificate chain/private key to use
```

Comment:
Tighter headline?

Reply:
Tightened this up to "Resolve a client certificate chain/private key to use as the client's identity.". I think from context it's clear enough that you make the resolution decision informed by the arguments, and the return type being `Option` gives enough of a hint that you can return `None`. We also spell out more detail about all of the above in the full comment past the headline.